packages/odyssey-react-mui/src/labs/TopNav.tsx

```diff
@@ -0,0 +1,459 @@
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import styled from "@emotion/styled";
+import { memo, useMemo, ReactElement, useCallback, KeyboardEvent } from "react";
+
+import type { HtmlProps } from "../HtmlProps";
+import {
+  AuraIcon,
+  AuraWordmarkIcon,
+  QuestionCircleIcon,
+  SettingsIcon,
+} from "../icons.generated";
+import { Link } from "../Link";
+import {
+  DesignTokens,
+  useOdysseyDesignTokens,
+} from "../OdysseyDesignTokensContext";
+
+export type TopNavLinkItem = {
+  id: string;
+  label: string;
+  /**
+   * link added to the nav item. if it is undefined, static text will be displayed.
+   * fires onClick event when it is passed
+   */
+  href?: string;
+  /**
+   * determines whether the link item is diabled
+   */
+  isDisabled?: boolean;
+  /**
+   * Event fired when the nav item is clicked
+   */
+  onClick?(): void;
+  /**
+   * The link target prop. e.g., "_blank"
+   */
+  target?: string;
+};
+
+export type UserProfileProps = {
+  /**
+   * Logged in user profile icon to be displayed in the top nav
+   */
+  profileIcon?: ReactElement;
+  /**
+   * Logged in user info to be displayed in the top nav
+   */
+  userName: string;
+  /**
+   * Org name of the logged in user
+   */
+  orgName: string;
+};
+
+export type TopNavProps = {
+  /**
+   * Determines whether to display the logo (aura & wordmark)
+   */
+  hasLogo?: boolean;
+  /**
+   *  Pass in a SearchField component with the variant="filled" prop set
+   */
+  SearchFieldComponent?: ReactElement;
+  /**
+   * Nav links in the top nav
+   */
+  topNavLinkItems: TopNavLinkItem[];
+  /**
+   * Pass in an additional componnet like Button that will displayed after the nav link items
+   */
+  AdditionalNavItemComponent?: ReactElement;
+  /**
+   * link to settings page
+   */
+  settingsLink?: string;
+  /**
+   * help link
+   */
+  helpLink?: string;
+  /**
+   * Displays user account info
+   */
+  userProfile?: UserProfileProps;
+} & Pick<HtmlProps, "testId">;
+
+const UserProfileContainer = styled("div", {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})(({ odysseyDesignTokens }: { odysseyDesignTokens: DesignTokens }) => ({
+  display: "flex",
+  alignItems: "center",
+  paddingRight: odysseyDesignTokens.Spacing4,
+}));
+
+const UserProfileIcon = styled("div", {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})(({ odysseyDesignTokens }: { odysseyDesignTokens: DesignTokens }) => ({
+  display: "flex",
+  paddingRight: odysseyDesignTokens.Spacing2,
+}));
```

Comment:
This isn't the icon itself right? It's the icon's container?

```suggestion
const UserProfileIconContainer = styled("div", {
  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
})(({ odysseyDesignTokens }: { odysseyDesignTokens: DesignTokens }) => ({
  display: "flex",
  paddingRight: odysseyDesignTokens.Spacing2,
}));
```